angularSTEP/src/app/signup/signup.component.ts

```diff
@@ -51,48 +51,56 @@ export class SignupComponent implements OnInit {
   //when form is submitted, if it is valid, checks if the username exists in Firestore. Then, if not,
   //creates a user and updates the "displayName" of the user authentication to the username (this is for the future
   //to make it easy to find user info since the doc Id is the username).
-  onSubmit() {
+  async onSubmit() {
```

Comment:
I'm guessing most of these are fixes that became more obviously needed in the test?